dns-record-check/src/adapter.ts

```diff
@@ -0,0 +1,43 @@
+import { Execute, DNSResponseAnswer, AdapterResponse } from '@chainlink/types'
+import { Validator } from '@chainlink/external-adapter'
+import { DNS } from '@chainlink/ea-factories'
+import { util } from '@chainlink/ea-bootstrap'
+import { DNSConfig } from '@chainlink/ea-factories/dist/factories/dns'
+
+const customParams = {
+  Answer: true,
+  record: true,
+}
+
+export const execute: Execute = async (input) => {
+  const validator = new Validator(input, customParams)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const { Answer, record } = validator.validated.data
+
+  const foundRecord = Answer.find((ans: DNSResponseAnswer) => ans.data.includes(record))
```

Comment:
This adapter does not get an `Answer` on input. It queries (delegates most of the work to) the underlying `dns-query` adapter to obtain an `Answer` and then tries to find a record.